modules/porous_flow/src/materials/PorousFlowCapillaryPressureLinear.C

```diff
@@ -0,0 +1,37 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "PorousFlowCapillaryPressureLinear.h"
+
+template<>
+InputParameters validParams<PorousFlowCapillaryPressureLinear>()
+{
+  InputParameters params = validParams<PorousFlowCapillaryPressureBase>();
+  params.addRequiredParam<Real>("pc_max",  "Maximum capillary pressure (Pa)");
+  params.addClassDescription("This Material provides a linear capillary pressure");
+  return params;
+}
+
+PorousFlowCapillaryPressureLinear::PorousFlowCapillaryPressureLinear(const InputParameters & parameters) :
+  PorousFlowCapillaryPressureBase(parameters),
+  _pc_max(getParam<Real>("pc_max"))
+{
+}
+
+void
+PorousFlowCapillaryPressureLinear::computeQpProperties()
+{
+  /// Capillary pressure and derivatives wrt phase saturation at the nodes
```

Comment:
Number of QPs = Number of nodes.  YES, this is a major assumption of our code and is necessary because of the full-upwinding and mass lumping that we require.  I'm not sure how to check this.  Could you tell me: (1) how to check it; (2) where you think it best be checked (i suspect in the Level 1 Materials - but in their constructors, or somewhere else?)